test/tests/types.test.js

```diff
@@ -406,95 +407,97 @@ describe('graphql - types parsing and validation', () => {
 
     describe('input literal', () => {
       test('cds.Decimal is correctly parsed from input literal float value', async () => {
-        const value = 123.45
-        const query = _getMutationForFieldWithLiteralValue(field, value, false)
-        const data = { TypesService: { MyEntity: { create: [{ [field]: String(value) }] } } }
-        const response = await POST('/graphql', { query })
+        const number = 123.45
+        const value = String(number)
+        const body = _getMutationForFieldWithLiteralValue(field, number, false)
+        const data = { TypesService: { MyEntity: { create: [{ [field]: value }] } } }
+        const response = await POST('/graphql', body)
         expect(response.data).toEqual({ data })
 
-        const result = await SELECT.from('sap.cds.graphql.types.MyEntity').columns(field)
-        expect(result).toContainEqual({ [field]: value })
+        const result = await SELECT.one.from('sap.cds.graphql.types.MyEntity').columns(field)
+        expect(String(result[field])).toEqual(value)
       })
 
       test('cds.Decimal is correctly parsed from input literal int value', async () => {
-        const value = 123
-        const query = _getMutationForFieldWithLiteralValue(field, value, false)
-        const data = { TypesService: { MyEntity: { create: [{ [field]: String(value) }] } } }
-        const response = await POST('/graphql', { query })
+        const number = 123
+        const value = String(number)
+        const body = _getMutationForFieldWithLiteralValue(field, number, false)
+        const data = { TypesService: { MyEntity: { create: [{ [field]: value }] } } }
+        const response = await POST('/graphql', body)
         expect(response.data).toEqual({ data })
 
-        const result = await SELECT.from('sap.cds.graphql.types.MyEntity').columns(field)
-        expect(result).toContainEqual({ [field]: value })
+        const result = await SELECT.one.from('sap.cds.graphql.types.MyEntity').columns(field)
+        expect(String(result[field])).toEqual(value)
       })
 
       test('cds.Decimal is correctly parsed from input literal numeric string value', async () => {
-        const value = 123.45
-        const query = _getMutationForFieldWithLiteralValue(field, value, true)
-        const data = { TypesService: { MyEntity: { create: [{ [field]: String(value) }] } } }
-        const response = await POST('/graphql', { query })
+        const number = 123.45
+        const value = String(number)
+        const body = _getMutationForFieldWithLiteralValue(field, number, true)
```

Comment:
```suggestion
        const body = _getMutationForFieldWithLiteralValue(field, value, true)
```
number is not a "numeric string value", so body does not contain a serialized string